CHANGELOG.md

```diff
@@ -9,6 +9,7 @@
 ### Removed
 
 ### Bug Fixes
+-  Remove TypeMeta declaration from the implementation of the objects
```

Comment:
We probably shouldn't classify this as a Bug Fix in the CHANGELOG since no user facing behavior changed and it's not clear what bug was fixed.
Also technically speaking, specifying the TypeMeta isn't a bug since we specified it correctly. Mostly just unnecessary and error prone as you mentioned.

Alternatively you can move this line to the CHANGED section since we updated the memcached controller template to not specify the TypeMeta. So more of a documentation update.

Also add a link to the PR since that's how we format the rest of the changelogs.
```suggestion
-  Remove TypeMeta declaration from the implementation of the objects [#1462](https://github.com/operator-framework/operator-sdk/pull/1462/)
```

Reply:
👍